Fix BigQuery tests for column names with special characters

ssheikin opened the pull request:
Some column names started to fail because of BigQuery internal change
~ between
18:00 16 of March 2023 GMT and
00:00 17 of March 2023 GMT,
first noticed when 90d6f0737a1f8dd4fbb8b85c2d95b20093d1b4ba merged to
master.

<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #dev in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description

Fixes #16603

<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

kokosing commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java:
```diff
@@ -659,6 +689,42 @@ public void testMissingWildcardTable()
+        catch (Exception e) {
+            throwable = e;
+        }
+        catch (Error e) {
```
error?

kokosing commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java:
```diff
@@ -659,6 +689,42 @@ public void testMissingWildcardTable()
+    {
+        Throwable throwable = null;
+        try {
+            super.testColumnName(columnName);
```
assertThatThrownBy?

kokosing commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java:
```diff
@@ -13,6 +13,7 @@
  */
```
I am a bit lost. Are we testing bigquery here? @ebyhr can you please let me know what is the point of test based on incorrect column names?

As I understand we should only update `filterColumnNameTestData`

ebyhr replied:
Testing unsupported column names helps us find SQL injection vector.

ebyhr commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java:
```diff
@@ -659,6 +689,42 @@ public void testMissingWildcardTable()
+    @Override
+    protected Optional<String> filterColumnNameTestData(String columnName)
+    {
+        if (NOT_ALLOWED_ANYMORE_COLUMN_NAMES.contains(columnName)) {
```
Creating and inserting into tables with these columns works and selecting the table isn't problem, right? This change hides the fact and we can't understand what happened after the operations. It would be better to deny such columns in all operations, or verify the result with different way, e.g. use query based API instead of storage API.

ssheikin commented:
@ebyhr hope I've addressed your comments. 
I think this PR is good enough to fix failing tests. However the additional work required to be able to SELECT from such columns again.
Please review.

ebyhr commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryAvroConnectorTest.java:
```diff
@@ -14,11 +14,32 @@
         extends BaseBigQueryConnectorTest
 {
+    // these column names started to fail because of BigQuery internal change ~ between 18:00 16 of March 2023 GMT and 00:00 17 of March 2023 GMT
+    // it's still possible to CREATE TABLE with such column names, and even INSERT some data, but it is not possible to SELECT from it.
```
> but it is not possible to SELECT from it.

This sentence is misleading. We can select the table using query based API. https://github.com/trinodb/trino/pull/16613#discussion_r1141524457

ebyhr commented on plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryAvroConnectorTest.java:
```diff
@@ -28,4 +49,39 @@ protected QueryRunner createQueryRunner()
+
+    // these column names started to fail because of BigQuery internal change ~ between 18:00 16 of March 2023 GMT and 00:00 17 of March 2023 GMT
+    // it's still possible to CREATE TABLE with such column names, and even INSERT some data, but it is not possible to SELECT from it.
+    // TODO is it possible to improve trino to be able to SELECT from such columns again?
```
Let's handle in follow-up https://github.com/trinodb/trino/issues/16664. We would like to fix CI failure first.

ssheikin commented:
@ebyhr Thank you for improving and merging
